### PR TITLE
Implementing to_native in lazyframe

### PR DESCRIPTION
**Pull request by HeitCunha**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/dataframe.py [hunk 1]
```diff
@@ -2602,10 +2603,10 @@ def collect(self) -> DataFrame[Any]:
             >>> lf
             ┌───────────────────────────────────────────────┐
             | Narwhals LazyFrame                            |
-            | Use `narwhals.to_native` to see native output |
+            | Use `df.to_native` to see native output |
             └───────────────────────────────────────────────┘
```
Comment (MarcoGorelli) [inline narwhals/dataframe.py:2607]: like this we're shortening the outupt, so the line at the right is no longer aligned - if you modify the message, you should also modify `__repr__`

maybe it should also say `DataFrame.to_native`? or `self.to_native`? `df` here doesn't refer to anything

**Review by MarcoGorelli — COMMENTED** [review]
thanks @HeitCunha for your pr! just a comment

File: narwhals/dataframe.py (hunk 1 above)
Comment (FBruzzesi) [reply narwhals/dataframe.py:2607]: May I suggest

>Use `.to_native` method to see native output

?

File: narwhals/dataframe.py [hunk 2]
```diff
@@ -2698,6 +2699,45 @@ def collect(self) -> DataFrame[Any]:
...
+            │ --- ┆ --- ┆ --- │
+            │ i64 ┆ f64 ┆ str │
+            ╞═════╪═════╪═════╡
+            │ 1   ┆ 6.0 ┆ a   │
+            │ 2   ┆ 7.0 ┆ b   │
+            │ 3   ┆ 8.0 ┆ c   │
+            └─────┴─────┴─────┘
+        """
+
+        return to_native(narwhals_object=self, strict=True)
```
Comment (FBruzzesi) [inline narwhals/dataframe.py:2739]: I know it was me suggesting this, but maybe as in #1020 and #1014, it is enough to:

```suggestion
        return self._compliant_frame._native_frame  # type: ignore[no-any-return]
```

**Review by MarcoGorelli — APPROVED** [review]
thanks @HeitCunha , and @FBruzzesi for review

**Pull request merged**